guides/traversing-pages.rst

```diff
@@ -1,164 +1,200 @@
 
-This object provides two very useful traversing methods:
+XPath Selector
+~~~~~~~~~~~~~~
 
-* ``find()`` - evaluates specific selector on the page content and returns
-  the last matched element or ``null``:
+The ``xpath`` selector type lets you use XPath queries to search elements
+in the page:
 
-  .. code-block:: php
+.. code-block:: php
 
-      $fieldElement = $page->find('named',
-          array('field', 'id|name|value|label')
-      );
-      $elementByCss = $page->find('css', 'h3 > a');
+    $anchorsWithoutUrl = $page->findAll('xpath', '//a[not(@href)]');
 
-* ``findAll()`` - evaluates specific selector on the page content and returns
-  an array of matched elements:
+.. caution::
 
-  .. code-block:: php
+    This selector searches an element inside the current node (which is ``<html>``
+    for the page object). This means that trying to pass it the XPath of
+    and element retrieved with ``ElementInterface::getXpath`` will not work
+    (this query includes the query for the root node). To check whether an
+    element object still exists in the browser page, use ``ElementInterface::isValid``
+    instead.
 
-      $fieldElements = $page->findAll('named',
-          array('field', 'id|name|value|label')
-      );
-      $elementsByCss = $page->findAll('css', 'h3 > a');
+.. _named-selector:
 
-Also, there's a bunch of shortcut methods:
+Named Selector
+~~~~~~~~~~~~~~
 
-* ``findById()`` - will search for an element by its ID
-* ``findLink()`` - will search for a link with ``link`` named selector
-* ``findButton()`` - will search for a button with ``button`` named selector
-* ``findField()`` - will search for a field with ``field`` named selector
+Named selectors provide a set of reusable queries for common needs. For conditions
+based on the content of elements, the named selector will try to find an
+exact match first. It will then fallback to partial matching in case there
+is no result for the exact match. The ``named_exact`` selector type can be
+used to force using only exact matching. The ``named_partial`` selector type
+can be used to apply partial matching without preferring exact matches.
 
-Nested Traversing
-~~~~~~~~~~~~~~~~~
+For the named selector type, the second argument of the ``find()`` method
+is an array with 2 elements: the name of the query to use and the value to
+search with this query:
 
-Every ``find*()`` method will return ``Behat\Mink\Element\NodeElement`` instance
-and ``findAll()`` will return an array of such instances. The fun part is
-you can make same old traversing on such elements too:
+.. code-block:: php
+
+    $escapedValue = $session->getSelectorsHandler()->xpathLiteral('Go to top');
+
+    $topLink = $page->find('named', array('link', $escapedValue);
+
+.. caution::
+
+    The named selector requires escaping the value as XPath literal. Otherwise
+    the generated XPath query will be invalid.
+
+The following queries are supported by the named selector:
+
+``id``
```

Comment:
Well, we have lots of changes between Mink 1.5 and 1.6 in the way function works (because they were not defined in a consistent way in Mink 1.5 because of the crappy driver tests). So making the doc highlight the changes from Mink 1.5 to 1.6 might make our job harder (and we are already missing lots of doc)